src/executables/util_service.c

```diff
@@ -1442,6 +1396,75 @@ process_service (int command_type, bool process_window_service)
   return status;
 }
 
+/*
+ * get_server_names -
+ *
+ * return:
+ *
+ *      out_buffer (out):
+ */
+static void
+get_server_names (char **name_buffer)
+{
+  FILE *input;
+  char buf[4096];
+  char cmd[PATH_MAX];
+
+  assert (name_buffer != NULL);
+
+  *name_buffer = NULL;
+
+  make_exec_abspath (cmd, PATH_MAX, (char *) UTIL_COMMDB_NAME " " COMMDB_ALL_STATUS);
+  input = popen (cmd, "r");
+  if (input == NULL)
+    {
+      return;
+    }
+
+  int offset = 0;
+  /* *INDENT-OFF* */
+  std::string delimiter = " ";
+  std::string result;
+  /* *INDENT-ON* */
+  while (fgets (buf, 4096, input) != NULL)
+    {
+      /* *INDENT-OFF* */
+      std::string s (buf);
+
+      /*
+      * ignore HA-applylogdb and HA-copylogdb
+      * check Server and HA-Server
+      */
+      std::string::size_type server_pos = s.find (CHECK_SERVER);
```

Comment:
It seems to be possible to check with HA-server with CHECK_SERVER.
Do you need to explicitly check the HA-server?

Reply:
Both "HA-Server" and "Server" have the string, "SERVER".  The purpose of this code is to get the name of the server after that space if there exists a string, "Server".